clients/benchmarks/client.cpp

```diff
@@ -1384,9 +1384,45 @@ int rocblas_bench_datafile(const std::string& filter,
                            const std::string& name_filter,
                            bool               any_stride)
 {
-    int ret = 0;
+    int                         ret = 0;
+    device_vector<rocblas_half> dA(267911424);
+    device_vector<rocblas_half> dB(267911424);
+    device_vector<rocblas_half> dC(267911424);
+    device_vector<rocblas_half> dD(267911424);
+    void*                       dAPointer = dA.device_vector_setup();
+    void*                       dBPointer = dB.device_vector_setup();
+    void*                       dCPointer = dC.device_vector_setup();
+    void*                       dDPointer = dD.device_vector_setup();
+
+    // Allocate host memory
+    host_vector<rocblas_half> hA(267911424);
+    host_vector<rocblas_half> hB(267911424);
+    host_vector<rocblas_half> hC(267911424);
+
+    // Check host memory allocation
+    CHECK_HIP_ERROR(hA.memcheck());
+    CHECK_HIP_ERROR(hB.memcheck());
+    CHECK_HIP_ERROR(hC.memcheck());
+
+    // Initialize host memory
+    // rocblas_check_matrix_type, uplo, host_vector, M, N, lda
+    rocblas_init_matrix_trig<rocblas_half>(rocblas_client_general_matrix, '*', hA, 16368, 16368, 1);
+    rocblas_init_matrix_trig<rocblas_half>(rocblas_client_general_matrix, '*', hB, 16368, 16368, 1);
+    rocblas_init_matrix_trig<rocblas_half>(rocblas_client_general_matrix, '*', hC, 16368, 16368, 1);
```

Comment:
The implementation looks like it's dependent on the arguments, so this data may not perfectly mimic the non-shared allocation version.